week2/materials/homework/webapp/src/App.js

```diff
@@ -1,17 +1,24 @@
-import Products from "./components/Products";
-import Header from "./components/Header";
-import useNotifications from "./hooks/useNotification";
+import { useEffect } from 'react';
+import Products from './components/Products';
+import Header from './components/Header';
+import useNotifications from './hooks/useNotification';
 
 function App() {
-  const { notifications } = useNotifications();
+	const { notifications, createNotification } = useNotifications();
 
-  return (
-    <div className="container">
-      <Header />
-      <h1 className="text-white">Good Green Groceries</h1>
-      <Products />
-    </div>
-  );
+	useEffect(() => {
```

Comment:
Instead of using useEffect, you can have a function in your hook to do the following two things: 

```javascript
//useNotifications.js
const addNotification(message) {
  SetNotifications(notifications.concat(“message”))
  // remove element with setTimeout
}

// rendering component.js
<div>{notifications.map(() => <div>...</div>)} 
```

Hope it makes sense, otherwise we will go through it in class.

Reply:
True, that's good idea. What I did was to display a green message when page refresh.